src/iosrv/init.c

```diff
@@ -211,13 +211,13 @@ dss_tgt_nr_get(int ncores, int nr)
 	 * if user requires more then uses the number calculated above
 	 * as creating more threads than #cores may hurt performance.
 	 */
-	if (nr >= 1 && nr < nr_default)
+	if (nr_default < nr)
+		D_PRINT("%d target XS(xstream) requested (#cores %d) exceeded "
+			"available XS (%d), Will try best effort to use %d "
+			"target XS\n", nr, ncores, nr_default, nr);
```

Comment:
Not sure what best effort means here. In practice, some xstreams will share the same core and this might not be good for performance, right? Then we should state this in this message.